test_obj_owner_in_obc_buckets is failing all the time on 4.18 - a fix is needed

Relevant runs: 

4.18: 
https://reportportal-ocs4.apps.dno.ocp-hub.prod.psi.redhat.com/ui/#ocs/launches/795/41667/1923638/1923717/log
https://reportportal-ocs4.apps.dno.ocp-hub.prod.psi.redhat.com/ui/#ocs/launches/795/42596/1960564/1960697/log
https://reportportal-ocs4.apps.dno.ocp-hub.prod.psi.redhat.com/ui/#ocs/launches/795/42598/1960915/1961048/log
https://reportportal-ocs4.apps.dno.ocp-hub.prod.psi.redhat.com/ui/#ocs/launches/795/42602/1961948/1962081/log
https://reportportal-ocs4.apps.dno.ocp-hub.prod.psi.redhat.com/ui/#ocs/launches/795/42604/1962630/1962763/log

Error: 
self = <ocs_ci.ocs.resources.namespacestore.NamespaceStore object at 0x7f34bfbe5c50>
timeout = 180, interval = 5


def verify_health(self, timeout=180, interval=5):
    """
    Health verification function that tries to verify
    a namespacestores's health until a given time limit is reached

    Args:
        timeout (int): Timeout for the check, in seconds
        interval (int): Interval to wait between checks, in seconds

    Returns:
        (bool): True if the bucket is healthy, False otherwise

    """
    log.info(f"Waiting for {self.name} to be healthy")
    try:

      for health_check in TimeoutSampler(
            timeout, interval, getattr(self, f"{self.method}_verify_health")
        ):

ocs_ci/ocs/resources/namespacestore.py:174: 



self = <ocs_ci.utility.utils.TimeoutSampler object at 0x7f34e31caf50>


def __iter__(self):
    if self.start_time is None:
        self.start_time = time.time()
    while True:
        self.last_sample_time = time.time()
        if self.timeout <= (self.last_sample_time - self.start_time):

          raise self.timeout_exc_cls(*self.timeout_exc_args)
E               ocs_ci.ocs.exceptions.TimeoutExpiredError: Timed out after 180s running oc_verify_health()



ocs_ci/utility/utils.py:1522: TimeoutExpiredError


During handling of the above exception, another exception occurred:


self = <mcg.test_obj_owner_md.TestObjOwnerMD object at 0x7f34de58d110>
mcg_obj = <ocs_ci.ocs.resources.mcg.MCG object at 0x7f34dcf56650>
bucket_factory = <function bucket_factory_fixture.._create_buckets at 0x7f34dda57420>
bucketclass = {'interface': 'OC', 'namespace_policy_dict': {'namespacestore_dict': {'ibmcos': [(1, None)]}, 'type': 'Single'}}
awscli_pod_session = <ocs_ci.ocs.resources.pod.Pod object at 0x7f34e39a7810>
test_directory_setup = SetupDirs(origin_dir='test_obj_owner_in_obc_buckets[ibmcos-nss]/origin', result_dir='test_obj_owner_in_obc_buckets[ibmcos-nss]/result')


@tier2
@skipif_disconnected_cluster
@skipif_proxy_cluster
@pytest.mark.parametrize(
    argnames=["bucketclass"],
    argvalues=[
        pytest.param(
            {
                "interface": "CLI",
                "backingstore_dict": {"aws": [(1, "us-east-2")]},
            },
        ),
        pytest.param(
            {
                "interface": "OC",
                "namespace_policy_dict": {
                    "type": "Single",
                    "namespacestore_dict": {"aws": [(1, "us-east-2")]},
                },
            },
        ),
        pytest.param(
            {
                "interface": "OC",
                "namespace_policy_dict": {
                    "type": "Single",
                    "namespacestore_dict": {"azure": [(1, None)]},
                },
            },
        ),
        pytest.param(
            {
                "interface": "OC",
                "namespace_policy_dict": {
                    "type": "Single",
                    "namespacestore_dict": {"ibmcos": [(1, None)]},
                },
            },
            marks=[skipif_kms_deployment],
        ),
    ],
    ids=[
        "aws-backingstore",
        "aws-nss",
        "azure-nss",
        "ibmcos-nss",
    ],
)
def test_obj_owner_in_obc_buckets(
    self,
    mcg_obj,
    bucket_factory,
    bucketclass,
    awscli_pod_session,
    test_directory_setup,
):
    """
    Test the ownership behavior of objects in OBC buckets.
    Objects in an OBC bucket should be owned by the noobaa operator
    account since the OBC was created by it.

    1. Create an OBC and upload some objects to it
    2. Verify that all the objets in the OBC bucket are owned by the noobaa operator account
    """
    # 1. Create an OBC and upload some objects to it

  bucket = bucket_factory(1, bucketclass=bucketclass)[0]

tests/functional/object/mcg/test_obj_owner_md.py:214: 



tests/conftest.py:3156: in _create_buckets
    else bucket_class_factory(bucketclass)
ocs_ci/ocs/resources/bucketclass.py:131: in _create_bucket_class
    namespacestores = namespace_store_factory(interface, nss_dict)
ocs_ci/ocs/resources/namespacestore.py:471: in _create_nss
    nss_obj.verify_health()



self = <ocs_ci.ocs.resources.namespacestore.NamespaceStore object at 0x7f34bfbe5c50>
timeout = 180, interval = 5


def verify_health(self, timeout=180, interval=5):
    """
    Health verification function that tries to verify
    a namespacestores's health until a given time limit is reached

    Args:
        timeout (int): Timeout for the check, in seconds
        interval (int): Interval to wait between checks, in seconds

    Returns:
        (bool): True if the bucket is healthy, False otherwise

    """
    log.info(f"Waiting for {self.name} to be healthy")
    try:
        for health_check in TimeoutSampler(
            timeout, interval, getattr(self, f"{self.method}_verify_health")
        ):
            if health_check:
                log.info(f"{self.name} is healthy")
                return True
            else:
                log.info(f"{self.name} is unhealthy. Rechecking.")
    except TimeoutExpiredError:
        log.error(
            f"{self.name} did not reach a healthy state within {timeout} seconds."
        )
        assert (

          False
        ), f"{self.name} did not reach a healthy state within {timeout} seconds."
E           AssertionError: ibmcos-ns-store-4ff8cec0ea5a4b1da184c819 did not reach a healthy state within 180 seconds.



ocs_ci/ocs/resources/namespacestore.py:187: AssertionError

--- a/ocs_ci/ocs/resources/namespacestore.py
+++ b/ocs_ci/ocs/resources/namespacestore.py
@@ -147,12 +147,16 @@ class NamespaceStore:
             bool: Based on whether the namespace store is healthy or not
 
         """
-        nss_data = OCP(
-            kind="namespacestore",
-            namespace=config.ENV_DATA["cluster_namespace"],
-            resource_name=self.name,
-        ).get()
-        return nss_data.get("status", {}).get("phase") == constants.STATUS_READY
+        try:
+            nss_data = OCP(
+                kind="namespacestore",
+                namespace=config.ENV_DATA["cluster_namespace"],
+                resource_name=self.name,
+            ).get()
+            return nss_data.get("status", {}).get("phase") == constants.STATUS_READY
+        except CommandFailed:
+            log.info(f"Namespacestore {self.name} not found or not accessible yet")
+            return False
 
     def verify_health(self, timeout=360, interval=5):
         """
@@ -190,7 +194,7 @@ class NamespaceStore:
                     f"{self.name} stuck in phase '{phase}', conditions: {conditions}"
                 )
             except Exception:
-                log.error(f"Failed to retrieve status for {self.name}")
+                log.exception(f"Failed to retrieve status for {self.name}")
             log.error(
                 f"{self.name} did not reach a healthy state within {timeout} seconds."
             )

--- a/ocs_ci/ocs/resources/namespacestore.py
+++ b/ocs_ci/ocs/resources/namespacestore.py
@@ -147,14 +147,12 @@ class NamespaceStore:
             bool: Based on whether the namespace store is healthy or not
 
         """
-        return (
-            OCP(
-                kind="namespacestore",
-                namespace=config.ENV_DATA["cluster_namespace"],
-                resource_name=self.name,
-            ).get()["status"]["phase"]
-            == constants.STATUS_READY
-        )
+        nss_data = OCP(
+            kind="namespacestore",
+            namespace=config.ENV_DATA["cluster_namespace"],
+            resource_name=self.name,
+        ).get()
+        return nss_data.get("status", {}).get("phase") == constants.STATUS_READY
 
     def verify_health(self, timeout=360, interval=5):
         """
@@ -180,6 +178,19 @@ class NamespaceStore:
                 else:
                     log.info(f"{self.name} is unhealthy. Rechecking.")
         except TimeoutExpiredError:
+            try:
+                nss_data = OCP(
+                    kind="namespacestore",
+                    namespace=config.ENV_DATA["cluster_namespace"],
+                    resource_name=self.name,
+                ).get()
+                phase = nss_data.get("status", {}).get("phase", "N/A")
+                conditions = nss_data.get("status", {}).get("conditions", [])
+                log.error(
+                    f"{self.name} stuck in phase '{phase}', conditions: {conditions}"
+                )
+            except Exception:
+                log.error(f"Failed to retrieve status for {self.name}")
             log.error(
                 f"{self.name} did not reach a healthy state within {timeout} seconds."
             )

